src/FactorGraph.jl

```diff
@@ -824,8 +824,10 @@ function doautoinit!(dfg::T,
         end
         pts,inferdim = predictbelief(dfg, vsym, useinitfct, logger=logger)
         setValKDE!(xi, pts, true, inferdim)
-        #TODO test
-        setVariablePosteriorEstimates!(xi)
+        # Update the estimates (longer DFG function used so cloud is also updated)
+        setVariablePosteriorEstimates!(dfg, xi.label)
+        # Update the data in the event that it's not local
+        mergeVariableSolverData!(dfg, xi)
```

Comment:
Perhaps this should rather be updateVariableSolverData! with a specific key. I’ll have to look in more detail, but merge didn’t work for the solved update as it could overwrite other keys.

Just as a breadcrumb, addSolverData(dfg, xi, :graphinit) for #615

Reply:
Okay, so I went through https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/wiki/Standard-Definitions-of-API-via-verbs-and-set-theory and looked at the functions (think they're still in flux), but I couldn't see the distinction between the two. Yes i think it should update just the initialized solver data, but it seems that the init method always just initializes the default key? Will update, but think we should clear up the distinctions

Reply:
Yes, I think currently it shouldn't make a difference.

Reply:
I agree with that this should be solveKey specific, since this is part of the current nonparametric graph initialization code.  Could we swap to verb `update` and ensure `solveKey=:default`.

I see verbNoun for `dest = merge*(src1,src2)` can be cleaned up a bit, and understand `merge` as creating a third collection but `merge*!(dest, src)` overwrites the first (the Julian way).

Reply:
@GearsAD , I made the change -- did you want to check it out before merging?